### ISSUE TITLE
package-directory-mismatch: false positive for main_test

### ISSUE THREAD

**Issue by wolfvictorCHI** [mgechev/revive#1532]
**Describe the bug**
revive's package-directory-mismatch reports a false positive for the usage of the package main_test on the test file main_test.go related to main.go.

**To Reproduce**
1. I updated revive `go install github.com/mgechev/revive@latest`
2. I run it with the following flags & configuration file:

```shell
revive -config myservice/reviveconfig.toml myservice/main_test.go
```

```toml
enableAllRules = true
```

**Expected behavior**
I expect package main_test to be accepted by revive without throwing a warning.

**Logs**
revive's output: `myservice/main_test.go:1:9: package name "main_test" does not match directory name "myservice"`.

**Desktop (please complete the following information):**
 - OS: macOS 15.6.1 (24G90)
 - Go 1.25.1

**Additional context**
revive seems to only detect this as an error when running it from outside the directory. If I run it on the directory, then it does not detect main_test as a violation of package-directory-mismatch.

**Comment by uloamaka** [comment]
Hello @chavacava I would like to work on this issue.

**Comment by uloamaka** [comment]
@chavacava  
While investigating this issue, I found that the package-directory-mismatch rule works as expected for correctly structured packages.

The behavior reported seems to result from using a mismatched directory and package name (for example, using package `main_test` inside a directory named `myservice`).

According to Go’s conventions, test packages are expected to match their directory names (e.g. `myservice_test` inside `myservice`), so the lint error is valid.

Changing the rule logic to allow such mismatches would risk weakening the intent of package-directory-mismatch, which is to enforce proper package-to-directory alignment.

The only reason why the `main_test` doesn't return the warning when ran from within its directory; is because it was actually skipped by assuming it directory was (.) hence ignored.

what do you suggest?

**TL;DR**
The reported behavior stems from a test misconfiguration, not from a rule bug.

@wolfvictorCHI 
You can change the package name to match its directory_name `myservice`, then it should not return that false-positive warning(myservice/main_test.go:1:9: package name "main_test" does not match directory name "myservice".).

**Comment by wolfvictorCHI** [comment]
Hi @uloamaka , thanks for looking into this. 

Could you please point me to specific Go convention that you are referring to? Before opening the issue, I did some research, did not find anything that specific and saw that [this pattern is used within the go source code](https://github.com/search?q=repo%3Agolang%2Fgo+%22package+main_test%22&type=code), so I assumed it to be valid. Also, is it expected that the tool behaves differently when run from within the directory?

**Comment by uloamaka** [comment]
Hi @wolfvictorCHI, thanks for the clarification.

After reviewing this again, I realized it was a misunderstanding on my part regarding the Go naming convention.

According to [Official Go documentation](https://go.dev/doc/effective_go#names:~:text=Another%20convention%20is%20that%20the%20package%20name%20is%20the%20base%20name%20of%20its%20source%20directory%3B%20the%20package%20in%20src/encoding/base64%20is%20imported%20as%20%22encoding/base64%22%20but%20has%20name%20base64%2C%20not%20encoding_base64%20and%20not%20encodingBase64.) and also explained in this [article](https://www.codementor.io/@inanc/understanding-go-packages-cgsx1hzrx#:~:text=Physically%2C%20a%20package%20is%20just%20a%20directory.%20By%20convention%2C%20in%20Go%2C%20directory%20names%20and%20the%20package%20names%20should%20be%20the%20same.%20Go%20tools%20and%20almost%20all%20of%20the%20Go%20community%20uses%20this%20convention.%20So%2C%20stick%20to%20it.) the package name is conventionally the same as the directory name, but main package are special case, hence `main_test` package should be treated differently.

And No it is not expected that the tools behaves differently when run from within the directory.

I’ll go ahead and resolve this behavior accordingly.

Thanks again for the clarification
